synapse/federation/transport/server.py

```diff
@@ -425,7 +425,18 @@ def on_POST(self, request):
             last_exception = None
             for invite in content["invites"]:
                 try:
-                    yield self.handler.exchange_third_party_invite(invite)
+                    if "signed" not in invite or "token" not in invite["signed"]:
+                        logger.info(
+                            "Discarding received notification of third party invite "
+                            "without signed: %s" % (invite,)
```

Comment:
Is discarding the invalid invites the right thing to do?
Should we be 400ing the identity server instead?